controller/controller.go

```diff
@@ -0,0 +1,82 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/filecoin-project/dealbot/config"
+	logging "github.com/ipfs/go-log/v2"
+
+	"github.com/gorilla/mux"
+	"github.com/pborman/uuid"
+)
+
+var log = logging.Logger("controller")
+
+type Controller struct {
+	server *http.Server
+	l      net.Listener
+	doneCh chan struct{}
+}
+
+func New(cfg *config.EnvConfig) (srv *Controller, err error) {
+	srv = new(Controller)
+
+	r := mux.NewRouter().StrictSlash(true)
+
+	// Set a unique request ID.
+	r.Use(func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			r.Header.Set("X-Request-ID", uuid.New()[:8])
+			next.ServeHTTP(w, r)
+		})
+	})
+
+	r.HandleFunc("/tasks", srv.getTasksHandler).Methods("GET")
+	//r.HandleFunc("/task", srv.getTaskHandler).Methods("GET")
```

Comment:
This should be a POST once the server starts serving tasks from a queue and tracking their status.